flow/layers/shader_mask_layer.cc

```diff
@@ -54,19 +54,32 @@ void ShaderMaskLayer::Paint(PaintContext& context) const {
       return;
     }
   }
+  auto shader_rect = SkRect::MakeWH(mask_rect_.width(), mask_rect_.height());
 
-  Layer::AutoSaveLayer save = Layer::AutoSaveLayer::Create(
-      context, paint_bounds(), cache_paint.sk_paint());
-  PaintChildren(context);
+  if (context.leaf_nodes_builder) {
+    context.builder_multiplexer->saveLayer(&paint_bounds(),
+                                           cache_paint.dl_paint());
+    PaintChildren(context);
 
-  SkPaint paint;
-  paint.setBlendMode(ToSk(blend_mode_));
-  if (shader_) {
-    paint.setShader(shader_->skia_object());
+    DlPaint dl_paint;
+    dl_paint.setBlendMode(blend_mode_);
+    if (shader_) {
+      dl_paint.setColorSource(shader_.get());
+    }
+    context.leaf_nodes_builder->translate(mask_rect_.left(), mask_rect_.top());
+    context.leaf_nodes_builder->drawRect(shader_rect, dl_paint);
```

Comment:
Looks like we forgot a restore here which made the rendering completely fail on internal tests.